.github/workflows/main.yml

```diff
@@ -20,36 +20,56 @@ jobs:
   pre-workflow-checks:
     runs-on: ubuntu-latest
     outputs:
-      should-run: ${{ steps.check.outputs.should-run }}
-      should-release: ${{ steps.tag.outputs.should-release }}
+      should-run: ${{ steps.set_outputs.outputs.should-run }}
+      image-names: ${{ steps.set_outputs.outputs.image-names }}
```

Comment:
@vnghia I have been trying to get this working for some time, but it seems that there is no way to publish a docker image from a PR (from a fork) since the `GITHUB_TOKEN` granted to the workflow only has read permissions (despite specifying write explicitly in the action).

Do you know a solution to this?

Reply:
So the current flow never works for PRs. Only for PRs from a branch in the same repo.

Reply:
let me think about it. you can use `env` instead of `secrets` but it will print your env in the log

Reply:
But to actually inject the secrets into env, i would still need access to secrets. Which are not available in PRs from forks.